File: android/src/main/java/com/getcapacitor/community/classes/events/FailToSendEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.community.BridgefyController.Reason;
import java.util.UUID;

public class FailToSendEvent extends FailToEvent {

    @NonNull
    UUID messageID;

    public FailToSendEvent(@NonNull UUID messageID, @NonNull Reason reason) {
        super(reason);
        this.messageID = messageID;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("messageID", messageID.toString());
        result.put("reason", super.toJSObject());

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Events/FailToSendEvent.swift
import Foundation
import Capacitor

@objc public class FailToSendEvent: FailToEvent {
    let messageID: UUID

    init(_ messageID: UUID, _ error: BridgefyError?) {
        self.messageID = messageID

        super.init(error)
    }

    override public func toJSObject() -> JSObject {
        var result = JSObject()

        result["messageID"] = messageID.uuidString.lowercased()
        result["reason"] = super.toJSObject()

        return result
    }
}
